app/src/main/java/com/example/music_app/ui/screens/search/SearchViewModel.kt

```diff
@@ -0,0 +1,118 @@
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+private const val LIMIT = 10
+
+@OptIn(FlowPreview::class)
+class SearchViewModel(
+    private val searchUseCase: RequestPlaylistsForSearchUseCase
+) : ViewModel() {
+
+    private var totalSize = 0
+    private var globalOffset = 0
+    private var currentText = ""
+
+    private val _isLoading: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val isLoading = _isLoading.asStateFlow()
+    private val _searchText = MutableStateFlow("")
+    val searchText = _searchText.asStateFlow()
+    private val _isSearching = MutableStateFlow(false)
+    val isSearching = _isSearching.asStateFlow()
+    private val _playlists = MutableStateFlow(listOf<Playlist>())
+    val playlists: StateFlow<List<Playlist>> = searchText
+        .debounce(300L)
+        .combine(_playlists) { text, _ ->
+            when {
+                text.isBlank() || text.length == 1 -> emptyList()
+                text == currentText -> _playlists.value
+                else -> {
+                    currentText = text
+                    _isSearching.value = true
+                    requestPlaylistsForSearch(
+                        text = text,
+                        isGetMorePlaylistsCase = false
+                    ).join()
+                    _playlists.value
+                }
+            }
+        }
+        .stateIn(
+            viewModelScope,
+            SharingStarted.WhileSubscribed(5000),
+            _playlists.value
+        )
+
+    private fun requestPlaylistsForSearch(
+        text: String,
+        isGetMorePlaylistsCase: Boolean,
+        offset: Int = 0
+    ) = viewModelScope.launch {
+        val playlistsResponseFlow = searchUseCase(
+            offset = offset,
+            limit = LIMIT,
+            prefix = text
+        )
+        playlistsResponseFlow.collect { playlistsResponse ->
+            playlistsResponse.onSuccess { playlists ->
+                if (isGetMorePlaylistsCase) {
+                    _playlists.value += playlists.playlists
+                    globalOffset += LIMIT
+                    _isLoading.value = false
+                } else {
+                    _playlists.value = playlists.playlists
+                    totalSize = playlists.totalSize
+                    globalOffset = LIMIT
+                    _isSearching.value = false
+                }
+            }
+        }
+    }
+
+    fun onSearchTextChange(text: String) {
+        _searchText.value = text
+    }
+
+    fun isScrollOnEnd(firstVisibleItemIndex: Int) {
+        if (firstVisibleItemIndex == (globalOffset - 6) && (totalSize - globalOffset) > 0) {
```

Comment:
Could you please extract 6 to the constants since it appears to be a magic number